### PR TITLE
Update hacking.md

### PR DESCRIPTION
**Pull request by miccal**
Closes https://github.com/caskroom/homebrew-cask/issues/28964 and https://github.com/caskroom/homebrew-cask/issues/17007.

Ping @caskroom/maintainers for suggestions and feedback - please feel free to edit my commit directly.

### CODE REVIEW

File: doc/development/hacking.md [hunk 1]
```diff
@@ -122,39 +87,48 @@ See [the relevant section in `adding_a_cask.md`](adding_a_cask.md#submitting-you
...
 
-We’re on IRC at `#homebrew-cask` on Freenode. If you are going to develop for Homebrew-Cask, it’s a great idea to hang out with us there. Here’s why:
+If you are going to develop for Homebrew-Cask, it’s a great idea to chat with us first. Here’s why:
 
-* Discuss your thoughts before coding and maybe get new ideas
+* Discuss your thoughts before coding by [opening an issue](https://github.com/caskroom/homebrew-cask/issues/new) and maybe get new ideas
 * Get feedback from the Travis-CI bot on build failures
-* Talk to [caskbot](https://github.com/passcod/caskbot) about checksums, version info, and releases
-* Just to be social!
+* Join us (and [caskbot](https://github.com/passcod/caskbot)) on IRC at `#homebrew-cask` on Freenode
```
Comment (reitermarkus) [inline doc/development/hacking.md]: Is anyone even using IRC? I only know I never used it.

File: doc/development/hacking.md [hunk 2]
```diff
@@ -15,8 +15,6 @@ Homebrew-Cask is designed to work like a traditional Unix tool:
 
 ## Project Status
 
-Homebrew-Cask is still young, and should be considered in alpha.
```
Comment (reitermarkus) [inline doc/development/hacking.md]: I think we should still keep this until we have `upgrade`.

Comment (miccal) [inline doc/development/hacking.md]: Agreed.

File: doc/development/hacking.md (hunk 1 above)
Comment (miccal) [inline doc/development/hacking.md]: Copied from README.md.

File: doc/development/hacking.md [hunk 3]
```diff
@@ -43,59 +41,28 @@ Casks, by contrast, only need to support the few installation methods used by ap
...
-  cd homebrew-cask
-  git remote add upstream https://github.com/caskroom/homebrew-cask.git
-  ```
-
-4. Now you have two copies of the Homebrew-Cask codebase on disk: the released version in `/usr/local/Library/Taps/caskroom/homebrew-cask`, and a development version in your private repo. To symlink the `Casks` and `rubylib` folders from `/usr/local/...` into your private repo, run the following script:
-
-  ```bash
-  /<path>/<to>/<private>/<repo>/developer/bin/develop_brew_cask
-  ```
+The setup is similar to that for contibuting to the core code of Homebrew - consult Homebrew's [documentation for maintainers]( https://github.com/Homebrew/brew/tree/master/docs#maintainers) for more information.
```
Comment (vitorgalvao) [inline doc/development/hacking.md]: Replace `- consult Homebrew's` with `— consult Homebrew’s` (correct typographic characters, I’ve changed them in all docs a while back).

Comment (vitorgalvao) [inline doc/development/hacking.md]: There’s an extraneous space after `](`.

File: doc/development/hacking.md [hunk 4]
```diff
@@ -122,39 +89,48 @@ See [the relevant section in `adding_a_cask.md`](adding_a_cask.md#submitting-you
...
 
-Be sure to run the test suite before submitting. If you forget, Travis-CI will do that for you and embarrass you in front of all your friends. :)
-
-You may also use a set of environment variables to increase verbosity:
+If no `<cask_token>`'s are given, then data for all Casks is returned. On failure, a blank line is returned on the standard output.
 
-* `TESTOPTS`, `TEST` etc. for the old [minitest suites](https://www.ruby-doc.org/stdlib-2.0.0/libdoc/rake/rdoc/Rake/TestTask.html)
-* `SPEC_OPTS`, `SPEC` etc. for [rspec suites](http://apidock.com/rspec/Spec/Rake/SpecTask)
-* `VERBOSE_TESTS` to see the standard output from the actual code = ignore the `shutup` helper
+For example, `brew cask _stanza appcast atom` outputs the `appcast` for the Cask [atom.rb](https://github.com/caskroom/homebrew-cask/blob/43ad9d8ddbad71fbeee42710d567861f080fedf8/Casks/atom.rb#L7), namely `https://github.com/atom/atom/releases.atom`.
```
Comment (vitorgalvao) [inline doc/development/hacking.md]: Should be instead “outputs the [`appcast` for the Cask atom.rb](https://github.com/caskroom/homebrew-cask/blob/43ad9d8ddbad71fbeee42710d567861f080fedf8/Casks/atom.rb#L7)” (notice how the link encompasses from the word `appcast`. That makes clearer we’re linking to that exact line.

File: doc/development/hacking.md [hunk 5]
```diff
@@ -122,39 +89,48 @@ See [the relevant section in `adding_a_cask.md`](adding_a_cask.md#submitting-you
...
 
 ```bash
-brew cask-tests
+brew cask _stanza <stanza_name> [ --table | --yaml | --inspect | --quiet ] [ <cask_token> ... ]
 ```
 
-Be sure to run the test suite before submitting. If you forget, Travis-CI will do that for you and embarrass you in front of all your friends. :)
-
-You may also use a set of environment variables to increase verbosity:
+If no `<cask_token>`'s are given, then data for all Casks is returned. On failure, a blank line is returned on the standard output.
```
Comment (vitorgalvao) [inline doc/development/hacking.md]: `<cask_token>`'s → `<cask_token>`’s (typography again).

File: doc/development/hacking.md [hunk 6]
```diff
@@ -122,39 +89,48 @@ See [the relevant section in `adding_a_cask.md`](adding_a_cask.md#submitting-you
...
 
-* `TESTOPTS`, `TEST` etc. for the old [minitest suites](https://www.ruby-doc.org/stdlib-2.0.0/libdoc/rake/rdoc/Rake/TestTask.html)
-* `SPEC_OPTS`, `SPEC` etc. for [rspec suites](http://apidock.com/rspec/Spec/Rake/SpecTask)
-* `VERBOSE_TESTS` to see the standard output from the actual code = ignore the `shutup` helper
+For example, `brew cask _stanza appcast atom` outputs the `appcast` for the Cask [atom.rb](https://github.com/caskroom/homebrew-cask/blob/43ad9d8ddbad71fbeee42710d567861f080fedf8/Casks/atom.rb#L7), namely `https://github.com/atom/atom/releases.atom`.
 
-Example of a very verbose output:
+The command `brew cask _stanza app --table --yaml alfred google-chrome adium voicemac logisim` outputs a human-readable data table of the `app` stanzas for the given Casks:
```
Comment (vitorgalvao) [inline doc/development/hacking.md]: I think we can remove this. `_stanza` is an internal advanced command anyway. It’s good to show it exists, but functionality should be studied by the interested.

File: doc/development/hacking.md [hunk 7]
```diff
@@ -122,39 +89,48 @@ See [the relevant section in `adding_a_cask.md`](adding_a_cask.md#submitting-you
...
+voicemac	---
+- VoiceMac/VoiceMac.app
+logisim	---
+- Logisim.app
 ```
 
-#### External Commands
-
-Advanced users may create their own external commands for Homebrew-Cask by following conventions similar to external commands for git or Homebrew. An external command may be any executable on your `$PATH` which follows the form `brewcask-<command>`. (So long as `<command>` does not conflict with an existing command verb.) The command will be invoked by `exec` and passed any unprocessed arguments from the original command-line. An external command may also be implemented as an executable Ruby file, on your `$PATH`, which follows the form `brewcask-<command>.rb`. The Ruby file will be `required` and will have full access to the Ruby environments of both Homebrew-Cask and Homebrew. Example external commands may be found in `developer/examples`.
+As a final example, the command `brew cask _stanza homepage > homepage_list.txt` will output the `homepage` of every Cask into the text file `homepage_list.txt`.
```
Comment (vitorgalvao) [inline doc/development/hacking.md]: [Same](https://github.com/caskroom/homebrew-cask/pull/29421/files#r98026358).

File: doc/development/hacking.md (hunk 1 above)
Comment (vitorgalvao) [inline doc/development/hacking.md]: I think this whole section can go away. We already mention it in the README, anyway.

Comment (vitorgalvao) [inline doc/development/hacking.md]: Last time I checked (a long time ago), there were still people on our IRC channel.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
Ideally, to close https://github.com/caskroom/homebrew-cask/issues/17007 we should also add `_stanza` do the `man` page. @reitermarkus already [made a section](https://github.com/Homebrew/brew/pull/1894/commits/2076b494a6233630025acd45d9a5766bd6bb785d), anyway, so it’d be a nice fit.

**Comment by miccal** [comment]
@vitorgalvao so would I update _both_ of these files:

https://github.com/Homebrew/brew/blob/master/manpages/brew-cask.1 
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/manpages/brew-cask.1.md

or just one and the other is automatically updated? I remember updating a `man` file in the past, and I _think_ I updated the wrong one?

PS: Thank you for the careful review and your updates.

File: doc/development/hacking.md (hunk 3 above)
Comment (miccal) [inline doc/development/hacking.md]: > Replace “- consult Homebrew's” with “— consult Homebrew’s” (correct typographic characters, I’ve changed them in all docs a while back).

How do you do that in Markdown?

File: doc/development/hacking.md (hunk 5 above)
Comment (miccal) [inline doc/development/hacking.md]: Same question - how do you do that? I always edit the .md files online, and I am not sure how to do this?

**Comment by vitorgalvao** [comment]
@miccal The `md` one is the one to update. The other is generated with `brew man`.

File: doc/development/hacking.md (hunk 3 above)
Comment (vitorgalvao) [inline doc/development/hacking.md]: Not a markdown feature, you have to actually type (or copy) the characters. On my (Portuguese) keyboard, <kbd>⌥</kbd><kbd>-</kbd> → — (em dash); <kbd>⇧</kbd><kbd>⌥</kbd><kbd>L</kbd> → ’ (apostrophe). I also have others committed to memory.

I think [this cheatsheet](https://www.typewolf.com/cheatsheet) has the US shortcuts.

I don’t mind making those changes, though. I like having the correct typographic characters.

Comment (miccal) [inline doc/development/hacking.md]: > I like having the correct typographic characters.

Me too, thanks for explaining how to do it.

**Comment by miccal** [comment]
@vitorgalvao done in https://github.com/Homebrew/brew/pull/1919.

I was not sure how much information to add to the `man` entry?

**Review by vitorgalvao — APPROVED** [review]

**Review by adidalal — APPROVED** [review]

**Pull request merged**